### PR TITLE
Improve tests for functional DAG

### PR DESCRIPTION
**Pull request by turbaszek**

### CODE REVIEW

File: plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py [hunk 1]
```diff
@@ -1,54 +1,113 @@
...
         assert isinstance(b, str)
 
+    def test_multiple_outputs_fail_without_type(self):
+        @task
+        def two_outputs():
+            return "Nights", "Ni!"
+
+        with pytest.raises(ValueError):
+            with DAG(dag_id="test_simple_build", default_args=default_args_test):
+                a, b = two_outputs()
```
Comment (turbaszek) [inline plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py]: This one is really interesting. @evgenyshulman how users should be aware that returning tuple requires proper type annotation?

File: plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py [hunk 2]
```diff
@@ -1,54 +1,113 @@
...
+        @task
+        def task1():
+            return True
+
+        with DAG(dag_id="test_simple_build", default_args=default_args_test) as dag:
+            bash_task = BashOperator(task_id="bash_task", bash_command="echo 'Ni!'")
+            bash_task >> task1()
+
+        _, b = dag.tasks
+        assert bash_task in b.upstream_list
```
Comment (turbaszek) [inline plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py]: This show that "functional ops" cannot be simply chained with other operators

File: plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py [hunk 3]
```diff
@@ -1,54 +1,113 @@
...
-    def test_simple_wiring(self):
-        with DAG(
-            dag_id="test_simple_build", default_args=default_args_test
-        ) as dag_operators:
-            a = t_A("check")
-            b = t_B(a)
+    def test_task_is_bind_to_dag(self):
+        @task
+        def sample_callable():
+            return True
```
Comment (turbaszek) [inline plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py]: I'm in favor of keeping as much in a test as we can. In this way, I don't have to check definitions from other files :) WDYT?

File: plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py [hunk 4]
```diff
@@ -1,54 +1,113 @@
...
-        assert result
-        a, b = result
+        @task
+        def two_outputs() -> Tuple[str, str]:
+            return "Nights", "Ni!"
+
+        with DAG(dag_id="test_simple_build", default_args=default_args_test):
+            a, b = two_outputs()
+
         assert isinstance(a, str)
```
Comment (evgenyshulman) [inline plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py]: let check that it's xcom values and not the real "nights" values

File: plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py (hunk 1 above)
Comment (evgenyshulman) [reply plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py]: +1, it sounds reasonable.

File: plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py (hunk 4 above)
Comment (turbaszek) [reply plugins/dbnd-airflow-operator/test_dbnd_airflow_operator/test_dag_build.py]: Done

**Comment by turbaszek** [comment]
Opened internal PR via #5

**Pull request closed without merging**